### PR TITLE
Unit/CalcAbsorbResist: Sanctified Wrath should pierce absorb effects

### PR DESCRIPTION
**Pull request by ghost**
by @P-Kito; retrieved from #15027

The paladin talent Sanctified Wrath should pierce absorb effects.

### CODE REVIEW

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -1647,6 +1647,9 @@ void Unit::CalcAbsorbResist(Unit* victim, SpellSchoolMask schoolMask, DamageEffe
 
     RoundToInterval(auraAbsorbMod, 0.0f, 100.0f);
 
+    auto absorb_ignoring_damage = CalculatePct(dmgInfo.GetDamage(), auraAbsorbMod);
```
Comment (jackpoz) [inline src/server/game/Entities/Unit/Unit.cpp]: https://github.com/TrinityCore/TrinityCore/pull/15027#discussion_r34158564

Comment (tkrokli) [inline src/server/game/Entities/Unit/Unit.cpp]: sure, I am open to suggestions for replacement code :memo:

Comment (Kittnz) [inline src/server/game/Entities/Unit/Unit.cpp]: He did @tkrokli

Comment (P-Kito) [inline src/server/game/Entities/Unit/Unit.cpp]: We can change it, of course. But what's the reason?

Comment (Runico) [inline src/server/game/Entities/Unit/Unit.cpp]: @P-Kito, I'm feel that auto use more resources to find out which type will  return by CalculatePct

Comment (jackpoz) [inline src/server/game/Entities/Unit/Unit.cpp]: auto is supposed to replace long-to-type types like std::vector<Creature*>::const_iterator, in this case instead it just replaces a simple uint32. it also makes more difficult to see if the type is a signed/unsigned integer or a float/double.

Comment (P-Kito) [inline src/server/game/Entities/Unit/Unit.cpp]: Alright. Sry.

Comment (tkrokli) [inline src/server/game/Entities/Unit/Unit.cpp]: @P-Kito : got a suggestion for how to replace the code in the line? I presume we will still be using
```cpp
[...] = CalculatePct(dmgInfo.GetDamage(), auraAbsorbMod);
```
but what to insert at the start of the line? Or is it OK just to remove `auto` like this:
```cpp
absorb_ignoring_damage = CalculatePct(dmgInfo.GetDamage(), auraAbsorbMod);
```

Comment (Nyeriah) [inline src/server/game/Entities/Unit/Unit.cpp]: Replace auto with uint32.

Comment (tkrokli) [inline src/server/game/Entities/Unit/Unit.cpp]: Thanks for the suggestion, I agree. Does it look OK if I name the variable `absorbIgnoringDamage` like this?
```cpp
    uint32 absorbIgnoringDamage = CalculatePct(dmgInfo.GetDamage(), auraAbsorbMod);
    dmgInfo.ModifyDamage(-absorbIgnoringDamage);
```
[...]
```cpp
    dmgInfo.ModifyDamage(absorbIgnoringDamage);

**Comment by tkrokli** [comment]
I suppose this PR does not contain anything useful, so I will close and remove it.

**Comment by jackpoz** [comment]
what makes you suppose so and what's the point of closing the PR ?

**Comment by tkrokli** [comment]
Sorry, I just had a rough day emotionally and made some bad judgements.

**Comment by jackpoz** [comment]
while this change might fix the issue, we have to make totally sure it doesn't break anything.
Unit::CalcAbsorbResist() is called for a lot of spells and only a very expert eye or hours of debugging and studying the sources can ensure the changes won't cause any issue. I personally have little knowledge of the spell system so I try to avoid these PRs and let others merge them.

After all the focus is on the quality of the code we provide to the users (seeking the "perfect commit" slows down the PR review and merging processes tho and that's a poor quality service to the contributors).

**Comment by Rushor** [comment]
@tkrokli - be like a bot and fuck on emotions.

**Comment by tkrokli** [comment]
Yes, I see. I would like to leave the academic discussion part to @P-Kito since he provided the code.
I just separated his PRs to help him split his original 5-in-1 PR.

**Comment by P-Kito** [comment]
I have been running this commit so far without any new bugs.
This code will only modify spells with: 
SPELL_AURA_MOD_TARGET_ABSORB_SCHOOL
or
SPELL_AURA_MOD_TARGET_ABILITY_ABSORB_SCHOOL

And afaik this only affects 3 or 4 spells then.
The old code was really badly implemented it reduced the absorb by a weird amount everytime one of these spells hit into absorb. (No wonder Retribution Paladin dealt no damage in PvP when he used wings)

**Comment by tkrokli** [comment]
This PR originated from the issue #3464 [**[Paladin] [Talent] Sanctified Wrath**](https://github.com/TrinityCore/TrinityCore/issues/3464) (already Closed)

**Comment by tkrokli** [comment]
This PR is quite small (+3 / -2 lines SLOC). Time to ship it? :) :shipit:

**Comment by P-Kito** [comment]
I have been using this for a long time now - No bugreports, no problems.

**Pull request merged**